.github/workflows/ci-prover-e2e.yml

```diff
@@ -24,17 +28,26 @@ jobs:
           echo "SCCACHE_GCS_RW_MODE=READ_WRITE" >> .env
           echo "RUSTC_WRAPPER=sccache" >> .env
           
-          mkdir -p prover_logs
+          mkdir -p prover_logs_${{matrix.compressor-mode}}
 
       - name: Start services
         run: |
           run_retried docker-compose -f ${RUNNER_COMPOSE_FILE} pull
           docker-compose -f ${RUNNER_COMPOSE_FILE} --profile runner up -d --wait
           ci_run sccache --start-server
+          ci_run git config --global --add safe.directory "*"
+
+      - name: Checkout DualVerifier commit
+        working-directory: ./contracts
+        if: matrix.compressor-mode == 'fflonk'
+        run: |
+          git fetch
+          git checkout b4d5b984
```

Comment:
Can you please comment on why we need to check this commit and for what purpose?